src/Features/CSharp/Portable/OrderModifiers/CSharpOrderModifiersHelper.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+
+namespace Microsoft.CodeAnalysis.CSharp.OrderModifiers
+{
+    internal static class CSharpOrderModifiersHelper
+    {
+        private static readonly char[] s_comma = { ',' };
+
+        /// <remarks>
+        /// Reference type so we can read/write atomically.
+        /// </remarks>
+        private static Tuple<string, Dictionary<int, int>> s_lastParsed;
+
+        public static bool TryGetOrComputePreferredOrder(string value, out Dictionary<int, int> preferredOrder)
```

Comment:
It's a temp value used only for the life of the operation.  Looking at ReadOnlyDictionary, it wraps an actual dictionary.  So every lookup goes through an indirection.  I'd really prefer the lowest overhead option here since we're going to hit every modifier of every item in the world :)